src/txn/txn_recover.c

```diff
@@ -548,6 +548,17 @@ __wt_txn_recover(WT_SESSION_IMPL *session)
     metafile = &r.files[WT_METAFILE_ID];
     metafile->c = metac;
 
+    /*
+     * We should check whether the history store file exists or not. or not. If it does not, then we
+     * should not apply rollback to stable to each table. This might happen if we're upgrading from
+     * an older version.
+     */
+    metac->set_key(metac, WT_HS_URI);
+    ret = metac->search(metac);
+    if (ret == WT_NOTFOUND)
+        hs_exists = false;
+    WT_ERR_NOTFOUND_OK(ret);
+
```

Comment:
I was considering doing a `reset` here to avoid pinning a page after I'm done checking. What do people think about this?

Reply:
I think it makes sense to reset the cursor here - we are done with it, and doing that makes it clear.
Otherwise this change LGTM

Reply:
a6fb04e